megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java

```diff
@@ -88,19 +98,96 @@ public EntitySprite(BoardView1 boardView1, final Entity entity,
     
     private String getAdjShortName() {
         if (onlyDetectedBySensors()) {
-            return Messages.getString("BoardView1.sensorReturn"); //$NON-NLS-1$
+            return Messages.getString("BoardView1.sensorReturn");
         } else {
-            String name = entity.getShortName();
-            int firstApo = name.indexOf('\'');
-            int secondApo = name.indexOf('\'', name.indexOf('\'')+1);
-            if ((firstApo >= 0) && (secondApo >= 0)) {
-                name = name
-                        .substring(firstApo+1, secondApo)
-                        .toUpperCase();
+            switch (GUIPreferences.getInstance().getUnitLabelStyle()) {
+            case FULL:
+                return standardLabelName();
+            case ABBREV:
+                if (entity instanceof Mech) {
+                    return entity.getModel();
+                } else {
+                    return abbreviateUnitName(standardLabelName());
+                }
+            case CHASSIS:
+                return reduceVehName(entity.getChassis());
+            case NICKNAME:
+                if (!pilotNick().isBlank()) {
+                    return "\"" + pilotNick().toUpperCase() + "\"";
+                } else if (!unitNick().isBlank()) {
+                    return "\'" + unitNick() + "\'";
+                } else {
+                    return reduceVehName(entity.getChassis());
+                }
+            case ONLY_NICKNAME:
+                if (!pilotNick().isBlank()) {
+                    return "\"" + pilotNick().toUpperCase() + "\"";
+                } else if (!unitNick().isBlank()) {
+                    return "\'" + unitNick() + "\'";
+                } else {
+                    return "";
+                }
+            default: // ONLY_STATUS
+                return "";
             }
```

Comment:
Please indent the switch. It's a minor thing, but I've found that it actually really helps with reading.